### PR TITLE
ez

### PR DESCRIPTION
**Pull request by frompotenza**

### CODE REVIEW

File: catkin_ws/src/auv_msgs/action/Quaternion.action
Comment (antoinedang) [inline catkin_ws/src/auv_msgs/action/Quaternion.action:1]: name Quaternion.action is not very clear, since it is both a quaternion and position action

File: catkin_ws/src/propulsion/src/thrust_mapper.py
Comment (antoinedang) [inline catkin_ws/src/propulsion/src/thrust_mapper.py:27]: Make sure this is updated with recent changes to the thrust mapper matrix, when I merge it will revert those changes if they are on noetic

File: catkin_ws/src/auv_msgs/action/Quaternion.action
Comment (antoinedang) [reply catkin_ws/src/auv_msgs/action/Quaternion.action:1]: or it should not need to have do_x, do_y, etc.

File: catkin_ws/src/planner/src/substates/quaternion_test.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/quaternion_test.py:1]: unnecessary once we're merging it in, controller should inherently use quaternion server for its rotation (and position?)

Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/quaternion_test.py:1]: the mission itself can stay but having a quaternion_action() function is not necessary

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/utility/controller.py:7]: do we still need all these imports if we're just using the quaternion server?

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/utility/controller.py:133]: unnecesary function mentioned above

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/utility/controller.py:130]: still unclear whether or not quaternion server does position as well as quaternion rotation. Wouldn't it be simpler and easier to read if there were 3 servers: quaternion, position, and effort?

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/utility/controller.py:24]: having all of these definitions for different combinations of true/false seems overly complicated, why not 5 global variables: do_x, do_y, do_z, do_rot, do_displace?

File: catkin_ws/src/planner/src/mission.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/mission.py]: leave the try/except statement, it is useful to kill missions without having to constantly handle keyboard exceptions at different levels

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:7]: We haven't fully transitioned from euler angles yet. Testing on clarke is necessary for us to deprecate the old code

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:133]: Yes because right now we are only implementing code for experimenting with quaternions. Switching the implementation of move and rotate would be too much right now

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:130]: For now we would still need a server that accommodates euler rotations until we can fully trust the quaternion code

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:24]: or nevermind I see now they're used to avoid defining them manually, but there has to be a better way. With superimposer + quaternion (I'm assuming we only have those 2), 6 DOFs for superimposer and 4 DOFs for quaternion, that's 6^2=36 options for superimposer, 4^2=16 options for quaternion, minus some because you can't have do_x on both at the same time, but you get my point

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [inline catkin_ws/src/planner/src/substates/utility/controller.py:51]: quaternion_server is a misleading name for a server that does position + quaternion

Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:51]: if we're merging it in it should have the same name StateServer as before, what happens inside shouldn't be reflected to planner

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:130]: if Quaternion does x,y,z and rotation then we really need to A. integrate the goal sending to Quaternion server into the controller interface functions and B. remove any trace of the old server in planner (we can keep the class definition as a backup or make a backup of the current state of noetic to a branch but having 3 servers in the controller is confusing af)

If not then it shouldn't take in a Pose and we should make that clearer

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:51]: Yeah its a bad name we can change it. But its unrealistic to think we're gonna replace the old stuff all at once. We need to merge temp code so we can test it before we can deprecate everything. I think we should merge it now and clean it up later. There isn't much point optimizing code that's still going to change

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:7]: we shouldn't have this intermediate state though, it's gonna be so messy and impossible to debug/test with

Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:7]: we should have this on a seperate branch and have no trace of euler angles on that branch, or save the euler angles version of noetic to another branch and remove any trace of them on noetic

Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:7]: I don't see how thats the case. We only added one function to the planner that you can just ignore

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:130]: > 

I would hold off on merging then until we are sure it works at least to a certain extent. Gonna be so messy otherwise

**Review by antoinedang — CHANGES_REQUESTED** [review]
I would only merge once we're 100% sure that this is the route we're taking and this branch has been cleaned up, it's more of a WIP rn and it would impede testing other things on noetic if we merged

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (MrMondrian) [reply catkin_ws/src/planner/src/substates/utility/controller.py:130]: We could do that but it would probably introduce like a million bugs

**Comment by MrMondrian** [comment]
> I would only merge once we're 100% sure that this is the route we're taking and this branch has been cleaned up, it's more of a WIP rn and it would impede testing other things on noetic if we merged

Me and Felipe told you guys this branch is a work in progress but everyone agreed to move the code in anyway

File: catkin_ws/src/planner/src/substates/utility/controller.py
Comment (antoinedang) [reply catkin_ws/src/planner/src/substates/utility/controller.py:51]: > 

The only advantage of merging right now that I see is to make it easier to test quaternions, and maybe during complex rotations (maybe?) avoiding Gimbal lock. The disadvantages are that we would have unstable and probably buggy code in the planner on the main branch that would make it much harder to test missions in the sim. I don't think it's a good idea to use the main branch as a temp merging branch, we should do that on a seperate branch no?

**Comment by MrMondrian** [comment]
I can make the names better but switching everything over will introduce bugs. This PR won't because it doesn't touch anything that the planner uses

**Comment by antoinedang** [comment]
> > I would only merge once we're 100% sure that this is the route we're taking and this branch has been cleaned up, it's more of a WIP rn and it would impede testing other things on noetic if we merged
> 
> Me and Felipe told you guys this branch is a work in progress but everyone agreed to move the code in anyway

Yeah I didn't realize it was this much a WIP, I really don't think it's a good idea to merge rn. Is there anything preventing us from having it seperate from noetic until it's ready?

**Comment by MrMondrian** [comment]
No we can keep it separate

**Comment by antoinedang** [comment]
> No we can keep it separate

ok bet let's do that imma close the PR if thats ok with u @MrMondrian

**Comment by MrMondrian** [comment]
That's fine

**Pull request closed without merging**
